### PR TITLE
OKTA-672843 include response for next() during paging operation

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
This change adds a parameter to api_response.next() which causes the response to be returned as the third tuple value during paging operations; this provides access to rate limit headers.

### CODE REVIEW

File: okta/api_response.py [hunk 1]
```diff
@@ -177,4 +183,4 @@ async def get_next(self):
                     self._request_executor, req, res_details, resp_body)
                 self._next = next_response._next
                 # yield next page
-                yield (next_response.get_body(), None)
+                yield (next_response.get_body(), None, next_response)
```
[the remark this answers is not in the document]
Comment (bryanapellanes-okta) [reply okta/api_response.py:186]: Good catch, thanks!

File: README.md [hunk 2]
```diff
@@ -925,6 +923,27 @@ loop = asyncio.get_event_loop()
...
+import asyncio
+
+async def main():
+    client = OktaClient()
+    users, resp, err = await client.list_users()
+    while True:
+        for user in users:
+            print(user.profile.login) 
+        if resp.has_next():
+            users, err, response = await resp.next(True) # Specify True to receive the response object for further analysis
```
Comment (justinabrokwah-okta) [inline README.md:939]: nit: `# Specify True to receive the response object as a third part of the tuple, for further analysis`

**Review by justinabrokwah-okta — APPROVED** [review]
🚀

**Review by justinabrokwah-okta — APPROVED** [review]

**Review by monde — APPROVED** [review]

**Comment by gabrielsroka** [comment]
so, when u first call list_users, it's `users, resp, err` and the 2nd time it's `users, err, resp` ???

also, in the example below (from the readme), it should be `resp`, not `response`

```py
from okta.client import Client as OktaClient
import asyncio

async def main():
    client = OktaClient()
    users, resp, err = await client.list_users()
    while True:
        for user in users:
            print(user.profile.login) 
        if resp.has_next():
            users, err, response = await resp.next(True) # Specify True to receive the response object as the third part of the tuple for further analysis
        else:
            break
```

**Comment by justinabrokwah-okta** [comment]
@gabrielsroka Yes, at least we don't break existing scripts for users but now we have the functionality for those who requested

**Comment by gabrielsroka** [comment]
There are no existing scripts that pass in `True`. Or you could just create a new function.

also, in the example (from the readme), should it be resp or response ?

**Comment by gabrielsroka** [comment]
also, since you're adding a new feature, maybe u could take another look at the 2-line PR i submitted last year, which @justinabrokwah-okta approved, but then got rolled back

https://github.com/okta/okta-sdk-python/pull/328

**Comment by gabrielsroka** [comment]
why is it includeResponse and not include_response?

**Comment by gabrielsroka** [comment]
> in the example (from the readme), should it be resp or response ?

to answer my own q, it looks like u need 2 different variables: resp and resp2. so you can't reuse the get_headers(), either. if u change resp2 to resp, it doesn't work (i get a weird error, it looks like dicts aren't being converted to objects -- idk if this is a bug)

```py
import okta.client
# import okta.api_response
import asyncio

async def main():
    # async with reuses the session, so it's much faster.
    async with okta.client.Client() as client:
        # Paginate users
        # resp: okta.api_response.OktaAPIResponse
        params = {'filter': 'profile.lastName eq "Doe"', 'limit': 1}
        users, resp, err = await client.list_users(params)
        print(resp.get_headers().get('x-rate-limit-remaining'))
        while True:
            for user in users:
                print(user.profile.login)
            if resp.has_next(): 
                users, err, resp2 = await resp.next(True)
                print(resp2.get_headers().get('x-rate-limit-remaining'))
            else: break

asyncio.run(main())
```

**Pull request merged**